receiver/dockerstatsreceiver/matcher.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package items contains common filtering logic that can be used to items
+// datapoints or various resources within other agent components, such as
+// monitors.  Filter instances have a Matches function which takes an instance
+// of the type that they items and return whether that instance matches the
+// items.
+package dockerstatsreceiver
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/gobwas/glob"
+)
+
+type Matcher interface {
+	Matches(string) bool
+}
+
+type StringMatcher struct {
+	standardItems       map[string]bool
+	anyNegatedStandards bool
+	regexItems          []regexItem
+	globItems           []globbedItem
+}
+
+// This utility defines a regex as
+// any string between two '/' characters
+// with the option of a leading '!' to
+// signify negation.
+type regexItem struct {
+	re        *regexp.Regexp
+	isNegated bool
+}
+
+func isRegex(s string) bool {
+	return len(s) > 2 && s[0] == '/' && s[len(s)-1] == '/'
+}
+
+type globbedItem struct {
+	glob      glob.Glob
+	isNegated bool
+}
+
+func isGlobbed(s string) bool {
+	return strings.ContainsAny(s, "*?[]{}!")
+}
+
+var _ Matcher = &StringMatcher{}
```

Comment:
nit: if you say
`var _ Matcher = (*StringMatcher)(nil)`
then you don't have to create a new struct instance

Reply:
updated